element_facemap/facial_behavior_estimation.py

```diff
@@ -0,0 +1,361 @@
+    package_version=''  : varchar(16)
+    """
+    
+    # Processing only the VideoRecordings that have their Info inserted.
+    @property
+    def key_source(self):
+        return FacemapTask & RecordingInfo.File
+
+    def make(self, key):
+        task_mode = (FacemapTask & key).fetch1('task_mode')
+
+        output_dir = (FacemapTask & key).fetch1('processing_output_dir')
+
+        if not output_dir:
+            output_dir = FacemapTask.infer_output_dir(key, relative=True, mkdir=True)
+            # update processing_output_dir
+            FacemapTask.update1({**key, 'processing_output_dir': output_dir.as_posix()})
+
+        if task_mode == 'trigger':
+            from facemap.process import run as facemap_run
+            facemap_params, motSVD, movSVD = (FacemapTask * FacemapParamSet & key).fetch1('params', 'do_mot_svd', 'do_mov_svd')
+
+            video_files = (FacemapTask * VideoRecording.File & key).fetch('file_path')
+            video_files = [find_full_path(get_facemap_root_data_dir(), video_file) for video_file in video_files]
+
+            facemap_run(video_files, motSVD=motSVD, movSVD=movSVD, proc=facemap_params, savepath=output_dir, parent=None)
+
+        _, facemap_dataset = get_loader_result(key, FacemapTask)
+        key = {**key, 'processing_time': facemap_dataset.creation_time}
+
+        self.insert1(key)
+
+
+@schema
+class FacialSignal(dj.Imported):
+    definition = """
+    # Facial behavioral variables estimated with Facemap
+    -> FacemapProcessing
+    """
+
+    class Regions(dj.Part):
+        definition = """
+        -> master
+        roi_id              : int               # Region no
+        ---
+        roi_name=''         : varchar(16)       # user-friendly name of the roi
+        xrange              : longblob          # 1d np.array - x pixel indices of the region
+        yrange              : longblob          # 1d np.array - y pixel indices of the region
+        xrange_bin          : longblob          # 1d np.array - binned x pixel indices of the region
+        yrange_bin          : longblob          # 1d np.array - binned y pixel indices of the region
+        """
+
+    class Vectors(dj.Part):
+        definition = """
+        -> master.Regions
+        ---
+        motsvd              : longblob          # 2d np.array - motion SVD for each region (nframes, components)
+        movsvd              : longblob          # 2d np.array - movie SVD for each region (nframes, components)
+        motmask_reshape     : longblob          # 3d np array - motion mask (y, x, components) - principle components
+        movmask_reshape     : longblob          # 3d np array - movie mask (y, x, components) - principle components
+        motion              : longblob          # 1d np.array - absolute motion energies across time (nframes)
+        """
+
+    class SingularValues(dj.Part):
+        # Diagonal elements of the square matrix in 1d
+        definition = """
```

Comment:
I think putting descriptions in the definition like this will render them as table titles if someone calls `schema.FacialSignal.SingularValues()`
```suggestion
        definition = """ # Diagonal elements of the square matrix in 1d
```

Reply:
I believe so.